source/adapters/level_zero/queue.cpp

```diff
@@ -955,6 +955,17 @@ ur_queue_handle_t_::ur_queue_handle_t_(
   ComputeCommandBatch.QueueBatchSize =
       ZeCommandListBatchComputeConfig.startSize();
   CopyCommandBatch.QueueBatchSize = ZeCommandListBatchCopyConfig.startSize();
+
+  static const bool useDriverCounterBasedEvents = [] {
```

Comment:
defining a lambda that only has one call site feels like extra complexity for little benefit.

Reply:
When it comes to querying environment variables, we define a lambda in all uses in the code. This provides consistency even if right now this is only used once.